types/ogre-tools-injectable.d.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+declare module "@ogre-tools/injectable" {
```

Comment:
Also, have you run any of these types through typescript? Because none of this compiles.

Reply:
Let me help! Surely enough, we don't get compilation errors. Can you post them?

Reply:
[Here is a link](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgSQBFgxgkATM7AT3yQCthsYEIkBhNmTZNOAbwBQcEYgZMYALjgAeYaJEAVOo2aYARgBs8oVBQDOBFRI3aZypPu5JcAGjiLipClQTB9APlvyFFq5htgbwUlJzJKGzd9YNEPAAofUWRVbi1gAGlgamkLFNMgxIBKOABeDzgAJWAYAFcoJEVqUnNjNTSAbQAiZH8UBExUToBdDwBueQBfQXlQSFgxVAwcPHxOJHQEAHM6-LCXG1pxZlYOLh4kNETdcMMiEnCqQ5STte5eKAFEqGBNhCs0aQJEJiPJpaT4VqpMwBajlAA+BE4AFtIBcUJD8jIYV4iqVygA3CAIcjjBRgb74sgwADKxOAAFF0OgJPpAcUynBCcTxokIJSoFA6dIWkcoToQHpyLcMWkRb07HBsfY-NZcPFEiJkiYwQ4ZdoYgo+WhBZQoNIqrV6o1mrltdouj1rCwBsBhjiFOyCUSSZNppqUGgsLhERAkUi2HrgOY9hFsFFykIFMSAPzSKyCpCbUmiTbVGOudyA8gIVMEfMHSEvM7vT0Octx9zZkSaBDMmjYbSlltt6gdhlIGpI7NTGbgaDwZCLIMrSNyXx0eUFXz1qJwa4GOAQdQpUoCCYG3WWJ39FhsAAKmCgmCR1T466lm+3El3-H38nK99uyPDSFnjnu+wNp4nxJkeATOqggIanAlDOLGUQ5Cu7gHv6vQQScF5Xjeiw5Aux4DBhl7XreUAHrWKoBLgw5+muY7zGQg5wN2TC9to9IDkiIGiPoyCbNoMBsAeMBXpYbgoAe+jYBApDkIoIk8VSMQjiIAD0KlrvoLYoAAtMW+j5NpFwStpWl4DUA76MA5DaQgKLjvoKlIBA2nmTUlnWfil76NpMD6KOczwFJR5wNg3wuq85x8CUcBxAAdPFuYwFUACONQIN8ry6KyipINQ7RDJ6iSrGwGzbFeaT1k8EhVigkVQKMQA)

Reply:
Hmm. That looks outdated, eg. `IComponentInjectable` is no more. This will go away today, when we relocate the types from lens to library `injectable`.

Reply:
Types relocated to library. No compilation errors present anymore.